syft/serde.py

```diff
@@ -1173,24 +1178,41 @@ def _detail_worker(worker: AbstractWorker, worker_tuple: tuple) -> pointers.Poin
     return referenced_worker
 
 
-def _simplify_GetNotPermittedError(error: GetNotPermittedError) -> tuple:
-    """Simplifies a GetNotPermittedError into its message"""
-    return (getattr(error, "message", str(error)),)
-
-
-def _detail_GetNotPermittedError(
-    worker: AbstractWorker, error_tuple: tuple
-) -> GetNotPermittedError:
-    """Details and raises a GetNotPermittedError
-
-    Args:
-        worker: the worker doing the deserialization
-        error_tuple: a tuple holding the message of the GetNotPermittedError
-    Raises:
-        GetNotPermittedError: the error thrown when get is not permitted
+def _simplify_exception(e):
     """
-
-    raise GetNotPermittedError(error_tuple[0])
+    Serialize information about an Exception which was raised to forward it
+    """
+    # Get information about the exception: type of error, error obj, traceback
+    tp, value, tb = sys.exc_info()
+    # Serialize the traceback
+    traceback_str = "Traceback (most recent call last):\n" + "".join(traceback.format_tb(tb))
+    # Include special attributes if relevant
+    if hasattr(e, "get_attributes"):
+        attributes = e.get_attributes()
+    else:
+        attributes = {}
+    return tp.__name__, traceback_str, _simplify(attributes)
+
+
+def _detail_exception(worker: AbstractWorker, error_tuple: Tuple[str, str, dict]):
+    """
+    Detail and re-raise an Exception forwarded by another worker
+    """
+    error_name, traceback_str, attributes = error_tuple
+    error_name, traceback_str = error_name.decode("utf-8"), traceback_str.decode("utf-8")
+    attributes = _detail(worker, attributes)
+    # De-serialize the traceback
+    tb = Traceback.from_string(traceback_str)
+    # Check that the error belongs to a valid set of Exceptions
+    if error_name in dir(sy.exceptions):
+        error_type = getattr(sy.exceptions, error_name)
+        error = error_type()
+        # Include special attributes if any
+        for attr_name, attr in attributes.items():
+            setattr(error, attr_name, attr)
+        reraise(error_type, error, tb.as_traceback())
```

Comment:
can we use raise instead of reraise?

Reply:
No you can't do this as far as I know, if you want to keep the stack trace which we want to. You are forced to use reraise